### PR TITLE
Upgrade operator-sdk to 1.22.0

### PR DESCRIPTION
**Pull request by iblancasa**
## Short description of the changes
- Update `operator-sdk` to the [latest version (1.22.0)
](https://github.com/operator-framework/operator-sdk/releases/tag/v1.22.0)

### CODE REVIEW

**Comment by iblancasa** [comment]
This PR is affected by #1962

**Comment by iblancasa** [comment]
@frzifus @rubenvp8510 this PR is ready for review :)

File: tests/e2e/miscellaneous/collector-autoscale/03-assert.yaml [hunk 1]
```diff
@@ -4,4 +4,4 @@ kind: Deployment
 metadata:
   name: simple-prod-collector
 status:
-  readyReplicas: 3
+  readyReplicas: 2
```
Comment (rubenvp8510) [inline tests/e2e/miscellaneous/collector-autoscale/03-assert.yaml:7]: Just curious why did you modify this parameter as part of the sdk upgrade?

**Review by rubenvp8510 — APPROVED** [review]
LGTM I just left one question

**Comment by pavolloffay** [comment]
The scorecard images need to be upgraded as well

**Comment by iblancasa** [comment]
@pavolloffay good catch! I'll do

**Comment by iblancasa** [comment]
They are since they depend on the `OPERATOR_SDK_VERSION` version.

```Make
SCORECARD_TEST_IMG ?= quay.io/operator-framework/scorecard-test:v$(OPERATOR_SDK_VERSION)
```

[Check this line from the log](https://github.com/jaegertracing/jaeger-operator/runs/7231492469?check_suite_focus=true#step:5:54).

**Comment by iblancasa** [comment]
Good catch! @pavolloffay  I'll do

**Comment by pavolloffay** [comment]
@iblancasa did you check if any golang dependencies need to be bumped? Usually operator-sdk version bump requires (and allows) to bump k8s related dependencies.

**Review by rubenvp8510 — APPROVED** [review]

**Comment by iblancasa** [comment]
@rubenvp8510 @rubenvp8510 could we merge this? Thanks!

**Pull request merged**